keps/498-GroupScheduling/README.md

```diff
@@ -0,0 +1,336 @@
+
+For statically provisioned clusters, we have recommended users use Kueue to handle group scheduling of JobSets once sufficient resources are available.
+
+However, for dynmaically provisioned clusters, we have no support for group scheduling in JobSet yet.
+
+### Goals
+
+- Enhance the Failure Policy API with options allowing the user to define a group scheduling policy.
+
+### Non-Goals
+
+- Upstream changes that would improve/simplify group scheduling support in JobSet.
+
+## Proposal
+
+### User Stories (Optional)
+
+#### Story 1: Group scheduling of all replicated jobs
+
+As a user, in order to make efficient use of expensive accelerator (GPU/TPU) resources, I use node
+auto-provisioning to provision infrastructure on an as-needed basis when a pending workload requires
+it, then deprovision it once it's no longer in use. Since the completion time of these provisioning
+operations is variable, I want to make sure JobSet pods which land on the earliest provisioned nodes
+do not start executing the main workload container until all the infrastructure is provisioned and
+all pods have started. Otherwise, the distributed initialization step will timeout in the earliest
+pods while waiting for the remaining provisioning to complete.
+
+#### Story 2: Group scheduling of a specific replicated job
+
+As a user, I have a JobSet with 2 replicated jobs:
+- `workers` which contains my primary batch workload, running on nodes with accelerator chips (GPUs)
+- `auxiliary` which contains auxiliary pods (proxy server, metrics service) running on CPU nodes.
+
+I want my batch workload workers to be scheduled as a group, but the auxiliary pods can start up
+individually at any time.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+### Proposed Group Scheduling API
+
+```go
+
+// JobSetSpec defines the desired state of JobSet
+type JobSetSpec struct {
+        …
+        // GroupSchedulingConfig defines the desired group-scheduling behavior
+        // for the JobSet. If unspecified, pods will begin running as soon as
+        // they are scheduled and ready.
+        GroupSchedulingConfig *GroupSchedulingConfig `json:groupSchedulingConfig`
+}
+
+// GroupSchedulingConfig defines the desired group-scheduling behavior for the
+// JobSet. If defined, the main workload container in pods part of the
```

Comment:
I don't understand this comment.

What is a main workload container?

Reply:
The container running the user workload (not the initContainer). 

Updated this to `user container` everywhere for clarity. Just trying to distinguish between the initContainer and user container.